tests/tabular_tests/records_tests/test_lists_factory.py

```diff
@@ -49,3 +50,13 @@ def test_lists_factory(data: DataObject,
     assert all(map(partial(table_record_is_valid,
                            table=table),
                    result))
+
+    unique_indices = [
+        index for index, column in enumerate(table.columns)
+        if is_column_unique(column)
+    ]
+
+    # Assert that all returned values are unique for the columns
+    # marked as unique
+    for index in unique_indices:
+      assert len({record[index] for record in result}) == len(result)
```

Comment:
can we extract this to a separate function like
```python
from sqlalchemy.schema import Table

from hypothesis_sqlalchemy.hints import RecordType
...
def records_satisfy_table_constraints(table_records: List[RecordType],
                                      *,
                                      table: Table) -> bool:
    unique_indices = [index 
                      for index, column in enumerate(table.columns)
                      if is_column_unique(column)]
    return all(len({record[index] for record in result}) == len(result)
               for index in unique_indices)
```
and move it to `tests.utils` module and then use it here like
```python
from tests.utils import records_satisfy_table_constraints
...
assert records_satisfy_table_constraints(records,
                                         table=table)
```